neevaco/neeva-android
# Change how the tasks are launched so that they appear in Android's Recents correctly

We've got three different Activities:
1) `FirstRunActivity`, which is used to show the First Run flow
2) `NeevaActivity`, which is used to hold the browser
3) `MainActivity`, which is used to launch the other two, based on whether the user has finished First Run before

After https://github.com/neevaco/neeva-android/pull/1037, we set the `taskAffinity=""`, which means that the tasks for each of the Activities don't stack anymore.  To make it work similarly to how it used to, this PR:
1) Excludes `MainActivity` from Recents so that we don't have it show up in Recents (clicking on it just opens the other Activity in a separate stack).
2) Makes `FirstRunActivity` and `NeevaActivity` be `singleInstance` instead of `singleTask`.
3) Makes each Activity launch other Activities explicitly in another Task, so that the `exclueFromRecents` flag on the `MainActivity` doesn't apply to the children Activities (which would incorrectly hide those Activities)
4) Make `FirstRunActivity` remove itself from Recents once it's done.

## Checklists

### How was this tested?

### Manual test cases
1) Cleared app data
2) Started the app
3) Checked that the app stayed in Recents by going home and launching a different app, then looking for Neeva in Recents
4) Re-foregrounded the app
5) Finished first run to get sent to the browser
6) Repeat step (3) to see that the app is still in Recents
7) Open a link in GMail to open the browser and see that the app shows up

## Issues addressed
* Fixes https://github.com/neevaco/neeva-android/issues/1040
